arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/raise/Fold.kt

```diff
@@ -224,17 +224,29 @@ public inline fun <Error, A> Raise<Error>.traced(
   @BuilderInference block: Raise<Error>.() -> A,
   trace: (trace: Trace, error: Error) -> Unit
 ): A {
-  val isOuterTraced = this is DefaultRaise && isTraced
-  val nested: DefaultRaise = if (isOuterTraced) this as DefaultRaise else DefaultRaise(true)
+  val nested = DefaultRaise(true)
   return try {
-    block.invoke(nested)
+    block(nested).also { nested.complete() }
   } catch (e: CancellationException) {
+    nested.complete()
```

Comment:
Good catch! Now I am wondering why those weren't there... 🤔
Almost feels like I would've done that on purpose, or some really bad regression happened here 😅

Reply:
I believe it's because previously this code would reuse a `DefaultRaise(true)` if it found it on the outside, and thus it didn't `complete()` it. Reusing the instance is what caused the CCE though, and so with that gone we can get better errors for leaks again.

Reply:
Right, thank you for clarifying ☺️